### PR TITLE
ci: skip tests involving L7 proxy if race detector is enabled

### LINKED ISSUE #13753: CI: Kafka & Cassandra policy tests consistently failing in Ginkgo-CI-Tests-runtime-Pipeline-Race-Detection
The following 4 tests are failing consistently with timeouts in Ginkgo-CI-Tests-runtime-Pipeline-Race-Detection:

- Suite-runtime.RuntimeKafka Kafka Policy Ingress
- Suite-runtime.RuntimeKafka Kafka Policy Role Egress
- Suite-runtime.RuntimeCassandra Tests policy allowing all actions
- Suite-runtime.RuntimeCassandra Tests policy disallowing Insert action

https://jenkins.cilium.io/view/Race%20Detection/job/Ginkgo-CI-Tests-runtime-Pipeline-Race-Detection/20/
https://jenkins.cilium.io/view/Race%20Detection/job/Ginkgo-CI-Tests-runtime-Pipeline-Race-Detection/19/
https://jenkins.cilium.io/view/Race%20Detection/job/Ginkgo-CI-Tests-runtime-Pipeline-Race-Detection/18/
https://jenkins.cilium.io/view/Race%20Detection/job/Ginkgo-CI-Tests-runtime-Pipeline-Race-Detection/17/
https://jenkins.cilium.io/view/Race%20Detection/job/Ginkgo-CI-Tests-runtime-Pipeline-Race-Detection/16/

Test result logs are too big to upload :cry:

### LINKED ISSUE #13757: CI: K8s Kafka/L7 policy tests consistently failing in race detector pipelines
The following two tests are failing consistently when build with the race detector enabled:

- Suite-k8s-1.18.K8sKafkaPolicyTest Kafka Policy Tests KafkaPolicies
- Suite-k8s-1.18.K8sPolicyTest Basic Test Validate CNP update Verifies that a CNP with L7 HTTP rules can be replaced with L7 Kafka rules

see e.g.

https://jenkins.cilium.io/view/Race%20Detection/job/Ginkgo-CI-Tests-4.19-Pipeline-Race-Detection/19/
https://jenkins.cilium.io/view/Race%20Detection/job/Ginkgo-CI-Tests-k8s1.12-Pipeline-Race-Detection/20/
https://jenkins.cilium.io/view/Race%20Detection/job/Ginkgo-CI-Tests-k8s1.19-Pipeline-Race-Detection/15/
https://jenkins.cilium.io/view/Race%20Detection/job/Ginkgo-CI-Tests-Pipeline-GKE-Race-Detection/20/ (this one has an additional failure in `Suite-k8s-1.16.K8sPolicyTest Basic Test Traffic redirections to proxy Tests HTTP proxy visibility without policy`)

This is similar to #13753, we should probably also skip these tests when the race detector is enabled.

### PR DESCRIPTION
**Pull request by tklauser**
Running tests involving the L7 proxy fail when using the race detector, so skip them in that case.

See individual commit for details.

Fixes #13753 
Fixes #13757

### CODE REVIEW

**Comment by tklauser** [comment]
test-race-runtime

**Comment by tklauser** [comment]
test-race-runtime

**Comment by tklauser** [comment]
test-me-please

**Review by aanm — APPROVED** [review]
Nice, really simple :D

**Comment by tklauser** [comment]
The tests somehow weren't disabled in Runtime-4.9-Race. It looks like `$(RACE)` is not passed to `ginkgo`, so the tests themselves are not built with the race detector enabled.

**Comment by tklauser** [comment]
test-race-runtime

**Comment by tklauser** [comment]
test-me-please

**Comment by tklauser** [comment]
test-me-please

**Comment by tklauser** [comment]
test-race-runtime

**Comment by tklauser** [comment]
test-race-gke

**Comment by tklauser** [comment]
test-race-4.19

**Comment by tklauser** [comment]
test-race-upstream-k8s

File: Documentation/contributing/testing/e2e.rst [hunk 1]
```diff
@@ -227,7 +229,7 @@ framework in the ``test/`` directory and interact with ginkgo directly:
       -cilium.timeout duration
             Specifies timeout for test run (default 24h0m0s)
 
-    Ginkgo ran 1 suite in 4.312100241s
+    Ginkgo ran 1 suite in 3.086476005s
```
Comment (christarazi) [inline Documentation/contributing/testing/e2e.rst]: Unnecessary change?

**Comment by tklauser** [comment]
test-race-runtime

**Review by christarazi — APPROVED** [review]
Nice. Just one non-blocking comment.

File: Documentation/contributing/testing/e2e.rst (hunk 1 above)
Comment (tklauser) [reply Documentation/contributing/testing/e2e.rst]: Side effect of copy&pasting the output :smile: I'll drop it if another revision is needed.

**Comment by tklauser** [comment]
test-race-gke

**Comment by tklauser** [comment]
test-race-4.19

**Comment by tklauser** [comment]
test-race-upstream-k8s

**Comment by tklauser** [comment]
test-me-please

**Review by aanm — APPROVED** [review]

**Comment by tklauser** [comment]
retest-net-next

**Comment by tklauser** [comment]
retest-4.9

File: Documentation/contributing/testing/e2e.rst (hunk 1 above)
Comment (tklauser) [reply Documentation/contributing/testing/e2e.rst]: Dropped the flag again as it's not necessary, we can just check the `RACE` env var at runtime.

**Comment by tklauser** [comment]
test-race-runtime

**Comment by tklauser** [comment]
test-me-please

**Review by pchaigno — APPROVED** [review]

**Comment by tklauser** [comment]
test-race-4.9

**Comment by tklauser** [comment]
test-race-4.19

**Comment by tklauser** [comment]
test-race-upstream-k8s

**Comment by tklauser** [comment]
retest-gke

**Comment by tklauser** [comment]
test-race-gke

**Comment by tklauser** [comment]
All required tests passed (with and without race detector), the only one failing is the k8s upstream test with race detector. This one also fails on master on the same test: https://jenkins.cilium.io/view/Race%20Detection/job/cilium-master-K8s-Upstream-Race-Detection/15/

Thus, marking as ready to merge.
